### ISSUE TITLE
Getting compilation error on opencv=1, Opencv 3.4.1 and 3.4.0 both installed 

### ISSUE THREAD

**Issue by ahsan856jalal** [AlexeyAB/darknet#945]
Hello @AlexeyAB 
pkg-config --modversion opencv 
gives me opencv 3.4.1 
whereas 
python-> import cv2;cv2.__version__
gives me 3.4.0 
During darknet compilation with opencv=1 , I am getting compilation errors which I believe is due to opencv 3.4.1
Can you please help me out ?
Regards
Ahsan

**Comment by jinyu121** [comment]
What is the error information? And try #946

**Comment by AlexeyAB** [comment]
@ahsan856jalal 
* Did you install OpenCV from this link? https://opencv.org/releases.html
* Or did you install OpenCV by using `pip`?
* Do you have several OpenCV versions?
* Did you specify path to the correct OpenCV version? https://github.com/AlexeyAB/darknet/issues/939

**Comment by ahsan856jalal** [comment]
Yes I used pip install and I think that installed opencv 3.4.1 and even though I am reinstalling 3.4.0 but somehow it is stucked with 3.4.1
Yes I have several versions , in python environment, it is 3.4.0 
and when I run command ' pkg-config --modversion opencv', it is showing me 3.4.1
I will try to change paths and will see whether it solves the problem

**Comment by ahsan856jalal** [comment]
I fixed the opencv version issue , i.e I have installed 3.4.0
but now I am getting this error

./src/http_stream.cpp:46:31: fatal error: /opencv2/opencv.hpp: No such file or directory
compilation terminated.
Makefile:133: recipe for target 'obj/http_stream.o' failed
make: *** [obj/http_stream.o] Error 1

**Comment by jinyu121** [comment]
I am using OpenCV 3.4.1, and can find `opencv.hpp` in `/usr/include/opencv2`. So please check if this file exists, or you should modify the path.

**Comment by AlexeyAB** [comment]
@ahsan856jalal  What OS do you use?

**Comment by dfsaw** [comment]
I have installed openCV 3.41.1. while running "make"  i am getting error

/home/affine/pyspark/anaconda3/include/opencv2/core/cvdef.h:485:1: error: unknown type name ‘namespace’
 namespace cv {
 ^
compilation terminated due to -Wfatal-errors.
Makefile:130: recipe for target 'obj/image.o' failed
make: *** [obj/image.o] Error 

Can you please help me out.

**Comment by AlexeyAB** [comment]
@dfsaw 

> I have installed openCV **3.41.1**. while running "make" i am getting error

OpenCV 3.4.1 has C API bug: https://github.com/AlexeyAB/darknet#you-only-look-once-unified-real-time-object-detection-versions-2--3

> This repository supports:
> 
> both Windows and Linux
> both OpenCV 2.x.x and OpenCV <= 3.4.0 **(3.4.1 and higher isn't supported)**
> both cuDNN v5-v7
> CUDA >= 7.5
> also create SO-library on Linux and DLL-library on Windows
